### ISSUE TITLE
Error adding metric: vttablet_v_replication_errors

### ISSUE THREAD

**Issue by artemvovk** [vitessio/vitess#7040]
#### Overview of the Issue

After manually recreating `MoveTables` workflow from an externally-managed keyspace (external) to a Vitess-managed keyspace: I am seeing excess of error logs in `vttablet` containers that manage external MySQL during VReplication:

```
vitess-cluster-vttablet-uscentral1c-1817714455-015018a1 vttablet E1114 16:16:50.968014       1 collectors.go:243] Error adding metric: Desc{fqName: "vttablet_v_replication_errors", help: "Errors during vreplication", constLabels: {}, variableLabels: [workflow type]}
```

The flow was to use one table from an external MySQL (5.6) and `Materialize` it into Vitess, then `SwitchReads` and `SwitchWrites` so that client applications will now read/write to Vitess instead of an existing/external MySQL.

The reason we did not use `MoveTables` is because it does not support transforms during initial `VReplication`, which  we required due to some bad `ENUM` type data in the original table (mistyped values are saved as `''` instead of erroring).

#### Reproduction Steps

Steps to reproduce this issue, example:

1. Deploy the following `vschema`:

```
GetVSchema recurly
{

}
GetVSchema external
{

}
```

1. Deploy the following `schema`:

```
    {
  "database_schema": "CREATE DATABASE /*!32312 IF NOT EXISTS*/ {{.DatabaseName}} /*!40100 DEFAULT CHARACTER SET utf8 */",
  "table_definitions": [
    {
      "name": "activities",
      "schema": "CREATE TABLE `activities` (\n  `id` bigint(20) unsigned NOT NULL AUTO_INCREMENT,\n  `site_id` bigint(20) unsigned NOT NULL,\n  `scope` enum('account_closure','account_updater','admin_console','api','data_retention','dunning','force_collect','ham','hosted_page','merchant','recurly_admin','recurly_js','recurly_system','recurring','token_api','transparent_post') DEFAULT NULL,\n  `actor_id` bigint(20) unsigned DEFAULT NULL,\n  `actor_type` enum('ApiKey','User','Services::MerchantIdentity::OauthClient') DEFAULT NULL,\n  `verb` varchar(50) DEFAULT NULL,\n  `acted_id` bigint(20) unsigned DEFAULT NULL,\n  `acted_type` enum('Account','Charge','CouponRedemption','GiftCard','Invoice','Subscription','SubscriptionChange','TaxLocationVerification','Transaction') DEFAULT NULL,\n  `meta` text,\n  `created_at` datetime(6) NOT NULL,\n  `account_id` bigint(20) unsigned DEFAULT NULL,\n  `request_id` varchar(50) DEFAULT NULL,\n  PRIMARY KEY (`id`),\n  KEY `index_activities_on_account_id` (`account_id`),\n  KEY `index_activities_on_acted_id_and_acted_type` (`acted_id`,`acted_type`),\n  KEY `index_activities_on_created_at` (`created_at`),\n  KEY `index_activites_on_site_id_and_acted_type` (`site_id`,`acted_type`),\n  KEY `index_activities_on_site_id_and_actor_id` (`site_id`,`actor_id`),\n  KEY `index_activities_on_site_id_and_created_at` (`site_id`,`created_at`),\n  KEY `index_activities_on_request_id` (`request_id`)\n) ENGINE=InnoDB DEFAULT CHARSET=utf8",
      "columns": [
        "id",
        "site_id",
        "scope",
        "actor_id",
        "actor_type",
        "verb",
        "acted_id",
        "acted_type",
        "meta",
        "created_at",
        "account_id",
        "request_id"
      ],
      "primary_key_columns": [
        "id"
      ],
      "type": "BASE TABLE",
      "data_length": "1089470464",
      "row_count": "13267433",
      "fields": [
        {
          "name": "id",
          "type": 778,
          "table": "activities",
          "org_table": "activities",
          "database": "recurly_qa",
          "org_name": "id",
          "column_length": 20,
          "charset": 63,
          "decimals": 0,
          "flags": 49699
        },
        {
          "name": "site_id",
          "type": 778,
          "table": "activities",
          "org_table": "activities",
          "database": "recurly_qa",
          "org_name": "site_id",
          "column_length": 20,
          "charset": 63,
          "decimals": 0,
          "flags": 53289
        },
        {
          "name": "scope",
          "type": 2074,
          "table": "activities",
          "org_table": "activities",
          "database": "recurly_qa",
          "org_name": "scope",
          "column_length": 64,
          "charset": 45,
          "decimals": 0,
          "flags": 256
        },
        {
          "name": "actor_id",
          "type": 778,
          "table": "activities",
          "org_table": "activities",
          "database": "recurly_qa",
          "org_name": "actor_id",
          "column_length": 20,
          "charset": 63,
          "decimals": 0,
          "flags": 49184
        },
        {
          "name": "actor_type",
          "type": 2074,
          "table": "activities",
          "org_table": "activities",
          "database": "recurly_qa",
          "org_name": "actor_type",
          "column_length": 156,
          "charset": 45,
          "decimals": 0,
          "flags": 256
        },
        {
          "name": "verb",
          "type": 6165,
          "table": "activities",
          "org_table": "activities",
          "database": "recurly_qa",
          "org_name": "verb",
          "column_length": 200,
          "charset": 45,
          "decimals": 0,
          "flags": 0
        },
        {
          "name": "acted_id",
          "type": 778,
          "table": "activities",
          "org_table": "activities",
          "database": "recurly_qa",
          "org_name": "acted_id",
          "column_length": 20,
          "charset": 63,
          "decimals": 0,
          "flags": 49192
        },
        {
          "name": "acted_type",
          "type": 2074,
          "table": "activities",
          "org_table": "activities",
          "database": "recurly_qa",
          "org_name": "acted_type",
          "column_length": 92,
          "charset": 45,
          "decimals": 0,
          "flags": 16640
        },
        {
          "name": "meta",
          "type": 6163,
          "table": "activities",
          "org_table": "activities",
          "database": "recurly_qa",
          "org_name": "meta",
          "column_length": 262140,
          "charset": 45,
          "decimals": 0,
          "flags": 16
        },
        {
          "name": "created_at",
          "type": 2064,
          "table": "activities",
          "org_table": "activities",
          "database": "recurly_qa",
          "org_name": "created_at",
          "column_length": 26,
          "charset": 63,
          "decimals": 6,
          "flags": 20617
        },
        {
          "name": "account_id",
          "type": 778,
          "table": "activities",
          "org_table": "activities",
          "database": "recurly_qa",
          "org_name": "account_id",
          "column_length": 20,
          "charset": 63,
          "decimals": 0,
          "flags": 49192
        },
        {
          "name": "request_id",
          "type": 6165,
          "table": "activities",
          "org_table": "activities",
          "database": "recurly_qa",
          "org_name": "request_id",
          "column_length": 200,
          "charset": 45,
          "decimals": 0,
          "flags": 16392
        }
      ]
    }
  ],
  "version": "02a13ecf3f2b8e6d144202fbbc8a5c88"
}
```

1. Run 
```
Materialize "{\"workflow\": \"fix_activities\", \"source_keyspace\": \"external\", \"target_keyspace\": \"recurly\", \"table_settings\": [{\"target_table\": \"activities\", \"source_expression\": \"select id, site_id, CASE scope WHEN ''  THEN NULL ELSE scope END as scope, actor_id, CASE actor_type WHEN ''  THEN NULL ELSE actor_type END as actor_type, verb, acted_id, CASE acted_type WHEN '' THEN NULL ELSE acted_type END as acted_type, meta, created_at, account_id, request_id from activities;\", \"create_ddl\": \"copy\"}]}"
```
1. Run
```
SwitchReads -tablet_type=replica recurly.fix_activities
```
1. Run
```
SwitchWrites recurly.fix_activities
```
1. At this  point there's a `_reverse` VReplication back into MySQL
1. Tail `vttablet` logs

#### Binary version
Build the sha from `release-8.0`:

```sh
vtgate --version
ERROR: logging before flag.Parse: E1114 16:40:25.686695      31 syslogger.go:149] can't connect to syslog
Version: 7e09d0c20 (Git branch 'HEAD') built on Wed Nov  4 18:02:13 UTC 2020 by vitess@a2a4c342bb7d using go1.13.15 linux/amd64

vttablet --version
ERROR: logging before flag.Parse: E1114 16:40:51.705127      32 syslogger.go:149] can't connect to syslog
Version: 7e09d0c20 (Git branch 'HEAD') built on Wed Nov  4 18:02:13 UTC 2020 by vitess@a2a4c342bb7d using go1.13.15 linux/amd64
```

#### Operating system and Environment details

Everything is running in GKE 1.17, is managed by `vitess-operator`

```
uname -a
Linux vitess-cluster-vttablet-uscentral1c-1817714455-015018a1 4.19.112+ #1 SMP Fri Sep 4 12:06:06 PDT 2020 x86_64 GNU/Linux
```

#### Log Fragments

```
I1114 16:42:04.579542       1 rpc_server.go:83] TabletManager.VReplicationExec(query:"select id, source, pos, stop_pos, max_replication_lag, state, db_name, time_updated, transaction_timestamp, message from _vt.vreplication where db_name = 'recurly_qa' and workflow = 'fix_activities_reverse'" )(on uscentral1c-1817714455 from ): (*tabletmanagerdata.VReplicationExecResponse)(nil)
I1114 16:42:04.587807       1 rpc_server.go:83] TabletManager.VReplicationExec(query:"select table_name, lastpk from _vt.copy_state where vrepl_id = 4" )(on uscentral1c-1817714455 from ): (*tabletmanagerdata.VReplicationExecResponse)(nil)
E1114 16:42:09.402264       1 collectors.go:243] Error adding metric: Desc{fqName: "vttablet_v_replication_errors", help: "Errors during vreplication", constLabels: {}, variableLabels: [workflow type]}
E1114 16:42:14.249637       1 collectors.go:243] Error adding metric: Desc{fqName: "vttablet_v_replication_errors", help: "Errors during vreplication", constLabels: {}, variableLabels: [workflow type]}
E1114 16:42:19.088416       1 collectors.go:243] Error adding metric: Desc{fqName: "vttablet_v_replication_errors", help: "Errors during vreplication", constLabels: {}, variableLabels: [workflow type]}
```

**Comment by rohit-nayak-ps** [comment]
Please verify if this PR https://github.com/vitessio/vitess/pull/7983 fixes this issue

**Comment by rohit-nayak-ps** [comment]
Fixed via #7983 and #8483
